armi/reactor/reactorParameters.py

```diff
@@ -770,3 +770,38 @@ def defineCoreParameters():
         )
 
     return pDefs
+
+
+def makeParametersReadOnly(r, readOnly=True):
+    """Convert all the paramters in a Reactor to read-only.
+
+    This method is pretty simple. It goes through all the children of a Reactor object,
+    recursively, and converts the parameters to read-only mode.
+
+    Parameters
+    ----------
+    r : Reactor
+        Full reactor object, to be modified.
+    readOnly : bool, optional
+        Make all the parameters read-only or not (writable).
+
+    Notes
+    -----
+    Once you make one Reactor read-only, it will lead to side effects in all Reactors you open in
+    ARMI. So if you run this function once with `readOnly=True`, and you want to move on to other
+    things, run this method on your reactor again with `readOnly=False` to unclock ARMI.
+    """
+    for pdef0 in r.p.paramDefs:
+        pdef0.readOnly = readOnly
+        for system in r.getChildren():
```

Comment:
One more comment I missed. This loop and it's children shouldn't be nested in the outer for loop.